src/Common/src/CoreLib/System/Memory.cs

```diff
@@ -77,15 +83,11 @@ public Memory(T[] array, int start, int length)
             _index = start;
             _length = length;
         }
-        
+
         // Constructor for internal use only.
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal Memory(OwnedMemory<T> owner, int index, int length)
         {
-            if (owner == null)
-                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.ownedMemory);
-            if (index < 0 || length < 0)
-                ThrowHelper.ThrowArgumentOutOfRangeException();
 
```

Comment:
Nit: Extra line

Reply:
It may be useful to add `// No validation performed; caller must provide any necessary validation.` comment here - similar to what it is in the other ctor.